frontend/src/actionCreators/parcelsActionCreator.spec.tsx

```diff
@@ -0,0 +1,86 @@
+import MockAdapter from "axios-mock-adapter";
+import axios from "axios";
+import {
+  fetchParcelDetail,
+  fetchParcels
+} from "actionCreators/parcelsActionCreator";
+import * as genericActions from "actions/genericActions";
+import * as API from "constants/API";
+import { IParcelListParams, IParcelDetailParams } from "constants/API";
+import * as MOCK from "mocks/dataMocks";
+import { ENVIRONMENT } from "constants/environment";
+
+const dispatch = jest.fn();
+const requestSpy = jest.spyOn(genericActions, "request");
+const successSpy = jest.spyOn(genericActions, "success");
+const errorSpy = jest.spyOn(genericActions, "error");
+const mockAxios = new MockAdapter(axios);
+
+beforeEach(() => {
+  mockAxios.reset();
+  dispatch.mockClear();
+  requestSpy.mockClear();
+  successSpy.mockClear();
+  errorSpy.mockClear();
+});
+
+describe("fetchParcels action creator", () => {
+
+  it("Null Params - Request successful, dispatches `success` with correct response", () => {
+    const url = ENVIRONMENT.apiUrl + API.PARCELS(null);
+    const mockResponse = { data: { success: true } };
+    mockAxios.onGet(url).reply(200, mockResponse);
+    return fetchParcels(null)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(successSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(6);
+    });
+  });
+
+  it("Request successful, dispatches `success` with correct response", () => {
+    const params: IParcelListParams = { neLat: 1, neLong: 2, swLat: 3, swLong: 4 };
+    const url = ENVIRONMENT.apiUrl + API.PARCELS(params);
+    const mockResponse = { data: { success: true } };
+    mockAxios.onGet(url).reply(200, mockResponse);
+    return fetchParcels(params)(dispatch).then(() => {
+      expect(requestSpy).toHaveBeenCalledTimes(1);
+      expect(successSpy).toHaveBeenCalledTimes(1);
+      expect(dispatch).toHaveBeenCalledTimes(6);
+    });
+  });
+
+  it("Request failure, dispatches `error` with correct response", () => {
```

Comment:
I'm wondering how much value a failure here is worth testing.  As I don't think the client is doing anything worth testing here.  Unless there is some functionality related to failures within the API.PARCELS(...) function.

Reply:
The point of this test is just to validate that an error response from the API triggers a dispatch to the redux error action.

Reply:
Ah so we're testing the Axios and Redux implementation of a error response.  Hmmm, still leaning towards not needing us write tests for their implementation unless we do something ourselves with error responses.

Reply:
Later on we'll wire up some UI to display errors from the store, but that would be a separate test.

Reply:
its not their implementation, we have a redux action that I wrote to handle writing errors to the store. I am testing that action has been written to the store after an error occurs.

Reply:
Makes sense